apps/discovery_api/lib/discovery_api_web/controllers/multiple_metadata_controller.ex

```diff
@@ -53,6 +54,34 @@ defmodule DiscoveryApiWeb.MultipleMetadataController do
       reraise e, __STACKTRACE__
   end
 
+  def advanced_search(conn, params) do
+    sort_by = Map.get(params, "sort", "name_asc")
+    query = Map.get(params, "query", "")
+
+    with {:ok, models} <- DatasetSearchIndex.search(query) do
```

Comment:
Would this be better as a case statement?

Reply:
It could, but I left it that way because this is only the first step and I didn't want to saddle the next person with turning it back into a with.

Reply:
That's what I was wondering. Thanks for explaining.

Reply:
Turns out Credo demands I do so in order to let the build pass